cypress/integration/home-spec.js

```diff
@@ -0,0 +1,130 @@
+        cy.get('.text-input').type('{backspace}')
+        cy.get('.poster-grid').children().should('have.length', 2)
+        cy.get('.text-input').type('{backspace}')
+        cy.get('.poster-grid').children().should('have.length', 3)
+    })
+    
+    it('When typing into the search input, the URL should reflect the value of the search.', () => {
+        cy.get('.text-input').type('mul')
+            .url().should('include', '?search=mul')
+    })
+})
+
+describe('Rancid Tomatillos sort flow', () => {
+
+    beforeEach(() => {
+        cy.fixture('movies.json').as('movies')
+        cy.intercept('GET',  'https://rancid-tomatillos.herokuapp.com/api/v2/movies', {
+            statusCode: 200,
+            ok: true,
+            body: { movies },
+        })
+        cy.visit('localhost:3000')
+    })
+    
+    it('When the user selects the filter for Rating: high to low, the movies should change accordingly.', () => {
+        cy.get('.ratings').select('descendingRating')
+        cy.get('.poster-grid').find('img').first()
+            .should('have.attr', 'src')
+            .and('equal', 'https://image.tmdb.org/t/p/original//uOw5JD8IlD546feZ6oxbIjvN66P.jpg')
+        cy.get('.poster-grid').find('img').last()
+            .should('have.attr', 'src')
+            .and('equal', 'https://image.tmdb.org/t/p/original//c59eplVELdwrUfGBUAZVin3HfaL.jpg')
+    })
+    
+    it('When the user selects the filter for Rating: low to high, the movies should change accordingly.', () => {
+        cy.get('.ratings').select('ascendingRating')
+        cy.get('.poster-grid').find('img').first()
+            .should('have.attr', 'src')
+            .and('equal', 'https://image.tmdb.org/t/p/original//c59eplVELdwrUfGBUAZVin3HfaL.jpg')
+        cy.get('.poster-grid').find('img').last()
+            .should('have.attr', 'src')
+            .and('equal', 'https://image.tmdb.org/t/p/original//uOw5JD8IlD546feZ6oxbIjvN66P.jpg')
+    })
+    
+    it('When the user selects the filter for Date: newest to oldest, the movies should change accordingly.', () => {
+        cy.get('.ratings').select('descendingDate')
+        cy.get('.poster-grid').find('img').last()
+            .should('have.attr', 'src')
+            .and('equal', 'https://image.tmdb.org/t/p/original//c59eplVELdwrUfGBUAZVin3HfaL.jpg')
+    })
+    
+    it('When the user selects the filter for Date: oldest to neweset, the movies should change accordingly.', () => {
+        cy.get('.ratings').select('ascendingDate')
+        cy.get('.poster-grid').find('img').first()
+            .should('have.attr', 'src')
+            .and('equal', 'https://image.tmdb.org/t/p/original//c59eplVELdwrUfGBUAZVin3HfaL.jpg')
+    })
+    
+    it('When the user selects a filter type for Randomize, the movies should change accordingly.', () => {
+        cy.get('.ratings').select('random')
+        cy.get('.poster-grid').children().should('have.length', 3)
+    })
```

Comment:
I am definitely curious if there is another way to go about this... I'm definitely going to think on it more, once my brain has had some rest.